kafka/consume.go

```diff
@@ -192,6 +240,41 @@ func (r *ConfluentConsumer) Close() error {
 	return r.c.Close()
 }
 
+// GetMeta return the confluence consumer metatdata
+func (r *ConfluentConsumer) GetMeta(allTopics bool, timeout time.Duration) (*kafkalib.Metadata, error) {
+	timeoutMs := timeout.Milliseconds()
+	if timeoutMs == 0 {
+		// Otherwise the call will be asynchronous, losing error handling.
+		return nil, errors.New("Timeout should be set when calling Seek")
+	}
+	if allTopics {
+		return r.c.GetMetadata(nil, true, int(timeoutMs))
+	}
+
+	return r.c.GetMetadata(&r.conf.Topic, false, int(timeoutMs))
+}
+
+// GetPartions retunrs the partition ids of a given topic
+func (r *ConfluentConsumer) GetPartions(timeout time.Duration) ([]int32, error) {
+	meta, err := r.GetMeta(false, timeout)
+	if err != nil {
+		return nil, err
+	}
+
+	return getPartitionIds(r.conf.Topic, meta)
+}
+
+// SetPartitionByKey sets the partition to consume messages from by the passed key
+func (r *ConfluentConsumer) SetPartitionByKey(key string, timeout time.Duration) error {
+	parts, err := r.GetPartions(timeout)
+	if err != nil {
+		return err
+	}
+	r.partition = GetPartition(key, parts)
```

Comment:
i understand this only works before you started consuming from kafka. it would be nice to have an atomic bool we flip once we assigned the partition and return an error here, if someone tries to set the partition afterwards

Reply:
Is there actually a way that we could remove this method and take an option in the Constructor?